docs/certificate support proposal.md

```diff
@@ -0,0 +1,229 @@
+In addition to Client secret, we'd want Microsoft.Identity.Web to support client certificates. The constraints are the following:
+- enable several ways of getting the certificate. You'd provide a description on how to get the certificate.
+  - from the certificate store (Windows) and a thumbprint ("440A5BE6C4BE2FF02A0ADBED1AAA43D6CF12E269")
+  - from the certificate store (Windows) and a distinguished name ("CN=TestCert")
+  - from a path on the disk (probably only for debugging locally)
+  - directly from a base64 representation of the certificate
+  - from a KeyVault address.
+- getting the certificate just in time, rather than paying the startup cost. For instance for a web app that signs in a user, do not load the certificate until an access token is needed to call a Web API.
+- when the certificate is stored in KeyVault, leverage Managed identity (probably though the Azure SDK for .NET)
+- help you rotating your certificates but letting you provide several (2) certificates
+
+## Decrypt certificates
+
+It's a different topic, but still touching on certificates: Web APIs can request token encryption (for privacy reasons). This is even compulsory for 1P Web APIs that access MSA identities. We also want to let you pass decryption certificates with the same flexibility as for client certificates.
+
+## Proposal 
+
+The current proposal is to have the following options to specify both client certificates and decrypt certificates from the configuration file.
+
+```Json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "Domain": "msidentitysamplestesting.onmicrosoft.com",
+    "TenantId": "7f58f645-c190-4ce5-9de4-e2b7acd2a6ab",
+    "ClientId": "86699d80-dd21-476a-bcd1-7c1a3d471f75",
+    "CallbackPath": "/signin-oidc",
+    "SignedOutCallbackPath ": "/signout-callback-oidc",
+
+    // To call an API
+    "ClientSecret": "[Copy the client secret added to the app from the Azure portal]",
+
+    // Exclusive with "ClientSecret"
+    "ClientCertificates": [
+      {
+        "CertificateFromThumbprint": {
+          "StoreLocation": "CurrentUser", // Optional, default = CurrentUser
+          "StoreName": "My",              // Optional, default = My
+          "certificateThumbprint": "440A5BE6C4BE2FF02A0ADBED1AAA43D6CF12E269"
+        }
+      },
+      {
+        "CertificateFromSubjectDistinguishedName": {
+          "StoreLocation": "CurrentUser", // Optional, default = CurrentUser
+          "StoreName": "My",              // Optional, default = My
+          "certificateSubjectDistinguishedName": "CN=TestCert"
+        }
+      },
+      {
+        "CertificateFromPath": "C:\Users\myname\Documents\TestJWE.pfx",
+        "CertificatePassword": "TestJWE"
+      },
+      {
+        "CertificateFromBase64Encoded": "Base64Encoded"
+      },
+      {
+        "CertificateFromKeyVault": "https://vaultname.vault.azure.net/certificates/certificateid/3fb1c62f74b844b0a2d9f1a3d289648d"
```

Comment:
I wonder if the naming here is a bit misleading if you expect to use managed identity here? AFAIK, you can only use this if the code is running in an Azure VM or an Azure Function?

Reply:
Actually no: Managed identities also work for App services, plenty of Azure services (increasing) and even locally on the dev box (using the shared token cache)